controller_interface/test/test_gps_sensor.cpp

```diff
@@ -46,16 +46,21 @@ struct GPSSensorTest : public testing::Test
     gps_sensor_name};
   std::vector<std::string> full_interface_names;
 
-  hardware_interface::StateInterface gps_state{
-    gps_sensor_name, gps_interface_names.at(0), &gps_states.at(0)};
-  hardware_interface::StateInterface gps_service{
-    gps_sensor_name, gps_interface_names.at(1), &gps_states.at(1)};
-  hardware_interface::StateInterface latitude{
-    gps_sensor_name, gps_interface_names.at(2), &gps_states.at(2)};
-  hardware_interface::StateInterface longitude{
-    gps_sensor_name, gps_interface_names.at(3), &gps_states.at(3)};
-  hardware_interface::StateInterface altitude{
-    gps_sensor_name, gps_interface_names.at(4), &gps_states.at(4)};
+  hardware_interface::StateInterface::SharedPtr gps_state =
+    std::make_shared<hardware_interface::StateInterface>(
+      gps_sensor_name, gps_interface_names.at(0), &gps_states.at(0));
+  hardware_interface::StateInterface::SharedPtr gps_service =
+    std::make_shared<hardware_interface::StateInterface>(
+      gps_sensor_name, gps_interface_names.at(1), &gps_states.at(1));
+  hardware_interface::StateInterface::SharedPtr latitude =
+    std::make_shared<hardware_interface::StateInterface>(
+      gps_sensor_name, gps_interface_names.at(2), &gps_states.at(2));
+  hardware_interface::StateInterface::SharedPtr longitude =
+    std::make_shared<hardware_interface::StateInterface>(
+      gps_sensor_name, gps_interface_names.at(3), &gps_states.at(3));
+  hardware_interface::StateInterface::SharedPtr altitude =
+    std::make_shared<hardware_interface::StateInterface>(
+      gps_sensor_name, gps_interface_names.at(4), &gps_states.at(4));
```

Comment:
```suggestion
  auto gps_state =
    std::make_shared<hardware_interface::StateInterface>(
      gps_sensor_name, gps_interface_names.at(0), &gps_states.at(0));
  auto gps_service =
    std::make_shared<hardware_interface::StateInterface>(
      gps_sensor_name, gps_interface_names.at(1), &gps_states.at(1));
  auto latitude =
    std::make_shared<hardware_interface::StateInterface>(
      gps_sensor_name, gps_interface_names.at(2), &gps_states.at(2));
  auto longitude =
    std::make_shared<hardware_interface::StateInterface>(
      gps_sensor_name, gps_interface_names.at(3), &gps_states.at(3));
  auto altitude =
    std::make_shared<hardware_interface::StateInterface>(
      gps_sensor_name, gps_interface_names.at(4), &gps_states.at(4));
```

Just to have it consistent wrt to other tests

Reply:
"auto is not allowed here", as they are class/struct members.